frontend/src/components/reviews/ReviewReply.tsx

```diff
@@ -0,0 +1,205 @@
+import { Autorenew as RefreshIcon, Send as SendIcon } from '@mui/icons-material';
+import { Dialog, Grid, IconButton, Stack, TextField, Tooltip, Typography, Zoom } from '@mui/material';
+import { random } from 'lodash';
+import { useEffect, useState } from 'react';
+import { ContentCopy as ClipboardIcon, Check as CopiedIcon } from '@mui/icons-material';
+import { GridLoader } from 'react-spinners';
+import { TypeAnimation } from 'react-type-animation';
+import { useQuery } from '@tanstack/react-query';
+import { Close as CloseIcon } from '@mui/icons-material';
+import { useTheme } from '@mui/material/styles';
+import { reviewService } from '@/services/review-service';
+
+const splitStringWithDelays = (longString: string, minDelay = 50, maxDelay = 300): Array<string | number> => {
+  const getRandomDelay = () => random(minDelay, maxDelay);
+  const getRandomLength = (minLength: number, maxLength: number) => random(minLength, maxLength);
+
+  const result = [];
+  let currentIndex = 0;
+
+  while (currentIndex < longString.length) {
+    const randomLength = getRandomLength(30, 50);
+    const substring = longString.slice(0, currentIndex + randomLength);
+
+    result.push(substring, getRandomDelay());
+    currentIndex += randomLength;
+  }
+
+  return result;
+};
+
+interface ReviewReplyProps {
+  reviewText: string;
+  open: boolean;
+  onClose: () => void;
+}
+
+const ReviewReply: React.FC<ReviewReplyProps> = ({ reviewText, open, onClose }) => {
+  const [writingReply, setWritingReply] = useState<boolean>(false);
+  const [previousReplies, setPreviousReplies] = useState<string[]>([]);
+  const [prompt, setPrompt] = useState<string>('');
+  const [copied, setCopied] = useState<boolean>(false);
+  const [typeAnimationKey, setTypeAnimationKey] = useState<number>(0);
+
+  const theme = useTheme();
+
+  const query = useQuery({
+    queryKey: ['generateReply', reviewText], // Unique query for each review
+    refetchOnMount: false,
+    enabled: open,
+    refetchOnWindowFocus: false,
+    queryFn: async () => {
+      const reply = await reviewService.generateReviewReply({ reviewText, prompt, previousReplies });
+      setPreviousReplies((prev) => [...prev, reply.text].slice(-10)); // Send only the latest 10 replies
+      setPrompt('');
+      setWritingReply(true);
+      return reply;
+    },
+  });
+
+  const { isFetching: loading, isError: error, data: reply } = query;
+  const replyText = reply?.text || '';
+
+  // Listen and rerender the TypeAnimation component
+  useEffect(() => {
+    setTypeAnimationKey((prev) => prev + 1);
+  }, [loading, writingReply]);
+
+  // Do not render anything when the dialog is closed
+  if (!open) {
```

Comment:
This looks like bad practice. Is it possible to conditionally render the component from the parent?